core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -428,7 +435,9 @@ public Session withDefaultProperties(Map<String, String> systemPropertyDefaults,
                 schema,
                 path,
                 traceToken,
-                timeZoneKey,
+                Optional.ofNullable(systemProperties.get(TIME_ZONE_ID))
```

Comment:
I'd add a comment that this is needed to be able to override a timezone with a WITH SESSION timezone = ''